core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -388,6 +388,21 @@ sealed abstract class Parser1[+A] extends Parser[A] {
   def rep1(min: Int): Parser1[NonEmptyList[A]] =
     Parser.rep1(this, min = min)
 
+  /** Use this parser to parse between values.
+    *
+    * Parses `b` followed by `this` and `c`.
+    * Returns only the values extracted by `this` parser.
+    */
+  def between[B, C](b: Parser[B], c: Parser[C]): Parser[A] =
+    b *> this <* c
```

Comment:
One minor item: `b *> (this <* c)` will be slightly more efficient since it will test b with less depth.

Reply:
actually, to geek out a bit more:
```
(b.void ~ (this ~ c.void)).map { case (_, (a, _)) => a }
```
is probably the most efficient since there is only one map, not two, one for each `<*` and `*>`.

Reply:
nice catch! let's do that and save some extra calls